src/Routes/RosPage/RosPage.js

```diff
@@ -162,10 +162,14 @@ class RosPage extends React.Component {
                                                     hasItems: true
                                                 });
                                                 return {
-                                                    results: results.data.map((system) => ({
-                                                        ...invSystems.find(({ id }) => id === system.inventory_id),
-                                                        ...system
-                                                    })),
+                                                    results: results.data.map((system) => {
+                                                        const invRec = invSystems.find(({ id }) => id === system.inventory_id);
+                                                        return ({
+                                                            ...invRec,
+                                                            ...(invRec ? { isDeleted: false } : { id: system.inventory_id, isDeleted: true }),
+                                                            ...system
+                                                        });
```

Comment:
This is not really reliable way of removing systems, much better approach would be
```JS
return {
  results: invSystems.map((system) => ({
    ...system
    ...results.find(({ inventory_id }) => inventory_id === system.id),
  }))
}
```

This way you'll map all inventory systems and add ROS system information to them.

Reply:
Hello @karelhala - 
Here I am not removing a system record but making it read only (i.e by adding plain text instead of link) using this `is_deleted` flag. Removal of such records will result into incorrect pagination because filtering & pagination happens on ROS backend side.